torch_xla/csrc/helpers.cpp

```diff
@@ -691,6 +755,109 @@ xla::XlaOp XlaHelpers::ImplicitBroadcast(xla::XlaOp op,
   return new_op;
 }
 
+std::pair<xla::XlaOp, xla::XlaOp>
+XlaHelpers::ImplicitBroadcastWithUnboundedDynamicShapes(
+    xla::XlaOp op1, xla::XlaOp op2, const xla::Shape& shape) {
+  XLA_CHECK(shape.is_unbounded_dynamic());
+
+  const xla::Shape& shape1 = ShapeHelper::ShapeOfXlaOp(op1);
+  const xla::Shape& shape2 = ShapeHelper::ShapeOfXlaOp(op2);
+  const auto& shape1_dims = shape1.dimensions();
+  const auto& shape2_dims = shape2.dimensions();
+  const auto& shape_dims = shape.dimensions();
+
+  XLA_CHECK(shape_dims.size() ==
+            std::max(shape1_dims.size(), shape2_dims.size()));
+
+  int64_t size_delta = shape2_dims.size() - shape1_dims.size();
+
+  // Extract the 'num_dims' counts of dimension sizes from the 'op', reshape
+  // them to 'tensor<1xi32>', and append them at 'op_dims'.
+  auto extract_dim_size = [](const xla::XlaOp op, const xla::Shape& op_shape,
+                             int num_dims, std::vector<xla::XlaOp>& op_dims) {
+    for (int i = 0; i < num_dims; i++) {
+      op_dims.push_back(xla::Reshape(xla::GetDimensionSize(op, i), {1}));
+    }
+  };
+
+  // Collect the dimension sizes of the broadcasted 'op1'.
+  // Example:
+  //   shape1 = [        1, ?_1]
+  //   shape2 = [?_2, ?_2',   5]
+  //    shape = [?_2,   ?,    ?]
```

Comment:
Wouldn't shape be:
`shape = [?_2, ?_2, 5]`

Reply:
done